### ISSUE TITLE
Option to select imagePullSecret only for CP or DP

### ISSUE THREAD

**Issue by RicHincapie** [istio/istio#51756]
**Describe the feature request**

If I set imagePullSecret in the helm chart values [here](https://github.com/istio/istio/blob/1c5963301ae491bdb7a76d9480fcc0610c045559/manifests/charts/base/values.yaml#L7) both Istiod (via its ServiceAccount) and all sidecars injected by it (default [template](https://github.com/istio/istio/blob/204da5ba47f295a8dc56936333e692f6a8707649/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml#L537)) would use that group of secrets.

In some use cases, users need to decide whether to use this secrets in the CP or in the CP + DP. This is currently left to the user customizing the template.

**Describe alternatives you've considered**

* Change injection template to check if the secret should not be injected like `index .ObjectMeta.Annotations "sidecar.istio.io/ignore-imagePullSecret"`. This lets current behavior as default

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by bleggett** [comment]
Yeah - this is a side effect of not consistently handling image property templating across all charts.

We support a global pull policy (fine) but don't have a local/scoped equivalent that isn't global for the same field.

Something like https://github.com/istio/istio/pull/51517/files#diff-cfe8054c550e36b443825fcdb9b37654018386b6bd3e38ea2d431f822f249e70R87 would fix this.

**Comment by deepakdeore2004** [comment]
can this be reopened?

**Comment by kanurag94** [comment]
Reopening this to work on this
